### PR TITLE
feat(colors): add new set for unified ramp of category colors in Teams theme

### PR DESCRIPTION
**Pull request by natashamayurshah**
Add all variants for the new set of [category colors](https://share.goabstract.com/ff44bfb8-840e-4396-91e4-750adb1e5efc?mode=design&sha=35ed81e812ac420a053175b09551bc973dd6e5e6).

### CODE REVIEW

File: packages/react/src/themes/teams-dark/colors.ts [hunk 1]
```diff
@@ -291,3 +291,40 @@ export const colorScheme: ColorSchemeMapping = {
...
+const createCategoryColorScheme = (color: string, customValues = {}) => {
+  return {
+    foreground1: categoryColors[color][250],
+    foreground2: categoryColors[color][550],
+    background: categoryColors[color][800],
+    ...customValues,
+  }
+}
+
+export const categoryColorSchemes: CategoryColorSchemeMapping = {
```
Comment (mnajdova) [inline packages/react/src/themes/teams-dark/colors.ts]: Let's preserve the previous naming convention and rename this to `categoryColorScheme`

File: packages/react/src/themes/teams-dark/colors.ts [hunk 2]
```diff
@@ -291,3 +291,40 @@ export const colorScheme: ColorSchemeMapping = {
     background: colors.yellow[100],
   }),
 }
+
+const createCategoryColorScheme = (color: string, customValues = {}) => {
+  return {
+    foreground1: categoryColors[color][250],
```
Comment (mnajdova) [inline packages/react/src/themes/teams-dark/colors.ts]: Let's replaced `foreground1` with `foreground` and `foreground2` with `foreground1`

Comment (mnajdova) [reply packages/react/src/themes/teams-dark/colors.ts]: Also, is there a plan for the future, more of these colors to be added in the color scheme? Not sure why we have so many values in the color palette, but we clearly only use a subset of them..

File: packages/react/src/themes/types.ts [hunk 3]
```diff
@@ -129,6 +137,39 @@ type EmphasisColorsStrict = Partial<{
 
 export type EmphasisColors = Extendable<EmphasisColorsStrict, ColorVariants>
 
+/**
+ * A type for a predefined category colors.
+ */
+type CategoryColorsStrict = Partial<{
```
Comment (mnajdova) [inline packages/react/src/themes/types.ts]: If this is supposed to be `Strict`, let's remove the `Partial` from the definition. If it is not suppose to be `Strict`, let's change the name to CategoryColors.

Comment (mnajdova) [reply packages/react/src/themes/types.ts]: Also these typings are specific to team theme, so we may move them to `packages/react/src/themes/teams/types.ts`

File: packages/react/src/themes/types.ts [hunk 4]
```diff
@@ -142,6 +183,14 @@ export type ColorValues<T, Colors extends string | number | symbol = ColorNames>
   T
 >
 
+/**
+ * A type for an extendable set of CategoryColorNames properties of type T
+ */
+type CategoryColorValues<
```
Comment (mnajdova) [inline packages/react/src/themes/types.ts]: `ColorValues` is already a generic, can't we just reuse it with different params

File: packages/react/src/themes/types.ts [hunk 5]
```diff
@@ -213,6 +262,17 @@ export type StrictColorSchemeMapping<
   default?: Scheme
 }
 
+export type CategoryColorScheme<T extends string | number | symbol = ComponentAreaName> = Partial<
+  Record<T, string>
+>
+
+export type CategoryColorSchemeMapping<
```
Comment (mnajdova) [inline packages/react/src/themes/types.ts]: We may also reuse the generics from the original `ColorSchemeMapping` type

File: packages/react/src/themes/teams-dark/colors.ts (hunk 2 above)
Comment (natashamayurshah) [reply packages/react/src/themes/teams-dark/colors.ts]: Moving the second comment discussion into the conversation with Sue Nguyen.

**Comment by mnajdova** [comment]
Final thing :) Let's add entry in the CHANGELOG.md file under `## [Unreleased]`'s `### Features` section, something like this:
- Add `categoryColors` and `categoryColorScheme` in Teams theme @natashamayurshah  ([#1711](https://github.com/stardust-ui/react/pull/1711))

**Review by mnajdova — APPROVED** [review]

**Pull request merged**